tests/contrib/gunicorn/test_gunicorn.py

```diff
@@ -175,6 +184,7 @@ def gunicorn_server(gunicorn_server_settings, tmp_path):
         SETTINGS_GEVENT_DDTRACERUN,
         SETTINGS_GEVENT_DDTRACERUN_MODULE_CLONE,
         SETTINGS_GEVENT_DDTRACERUN_DEBUGMODE_MODULE_CLONE,
+        SETTINGS_GEVENT_SPANAGGREGATOR_RLOCK,
```

Comment:
does this reproduce the issue without the RLock?

what relation does this gevent test have to the change? The fix is listed for sqlalchemy, do we have a sqlalchemy test case that replicates this issue?

Reply:
No, I haven't been able to recreate it in a test unfortunately. I mostly added it for fear that changing the lock might cause issues with Gunicorn.

Reply:
I was able to repro manually... but it involved forcing a span creation during `on_span_start` of `SpanAggregator`... but that isn't really something we can recreate in a test?